events/src/camera.rs

```diff
@@ -1,15 +1,19 @@
 use serde::{Deserialize, Serialize};
+use url::Url;
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct Event {
     pub cameras: Vec<Camera>,
 }
 
 /// `Camera` type represents different camera types: IP, USB, CSI cameras
-#[derive(Serialize, Deserialize, Default, Debug, Clone)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct Camera {
-    /// ONVIF address if camera is ONVIF-compliant
-    pub onvif_addr: Option<String>,
+    /// URI to access the camera.
+    ///
+    /// - USB or CSI camera: local file name (for example `file:///dev/video0`).
+    /// - IP camera: ONVIF device management service URL (for example `http://192.168.1.2/device`);
```

Comment:
Kudos for add docs with examples!

Reply:
The commit log says "Previously `uri` was used to point to" but that has remained unchanged so it's confusing. Also,:

> and there's inconsistency here: for local cameras we
> can use `uri` to access camera itself but for IP cams
> we cannot.

I was also very confused by this but then I realized you are talking about the limitation of the API here and not the the cameras themselves. So I'd reword this.

Reply:
I wouldn't say it remained unchanged because I changed the comment :)

> limitation of the API

Yep, it's about API of course, will reword.